src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -35,8 +37,27 @@ private static async Task<int> Egress<TOptions>(EgressProvider<TOptions> provide
             EgressArtifactResult result = new();
             try
             {
-                string jsonConfig = Console.ReadLine();
-                ExtensionEgressPayload configPayload = JsonSerializer.Deserialize<ExtensionEgressPayload>(jsonConfig);
+                StdInStream = Console.OpenStandardInput();
+
+                const int payloadLengthBufferSize = sizeof(long); // Stream's Position is a long
+
+                byte[] payloadLengthBuffer = new byte[payloadLengthBufferSize]; 
+
+                if (payloadLengthBufferSize != await StdInStream.ReadAsync(payloadLengthBuffer))
+                {
+                    throw new EgressException(GenericEgressFailureMessage);
+                }
+
+                long payloadBufferSize = BitConverter.ToInt64(payloadLengthBuffer);
+
+                byte[] payloadBuffer = new byte[payloadBufferSize];
+
+                if (payloadBufferSize != await StdInStream.ReadAsync(payloadBuffer))
```

Comment:
Yep, GitHub is acting up today for some reason...I didn't see any of his comments until _after_ I put up the first set of changes unfortunately. This will be in the next version.